core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +315,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput') {
+        var maxSelectionCount = (
+            customizationArgs.maxAllowableSelectionCount.value);
+        if (maxSelectionCount === 1) {
+          var numChoices = $scope.getAnswerChoices().length;
+          // Use array to store booleans that become true
+          // if answerChoice is covered
+          handledAnswersArray = [];
+          for (var i = 0; i < numChoices; i++) {
+            handledAnswersArray.push(false);
+            answerChoices.push($scope.getAnswerChoices()[i].val);
+          }
+
+          var newHandledAnswers = function(current) {
+            return handledAnswersArray.map(function(element, index) {
+              if (index !== current || element === true) {
+                return true;
+              } else if (index === current && element !== true) {
+                return false;
+              }
+            });
+          };
+
+          answerChoices.forEach(function(answerChoice, choiceIndex) {
+            answerGroups.forEach(function(answerGroup) {
+              var ruleSpecs = answerGroup.rule_specs;
+              ruleSpecs.forEach(function(ruleSpec) {
+                var ruleInputs = ruleSpec.inputs.x;
+                ruleInputs.forEach(function(ruleInput) {
+                  if (ruleInput === answerChoice) {
+                    if (ruleSpec.rule_type === 'Equals') {
+                      handledAnswersArray[choiceIndex] = true;
+                    } else if (ruleSpec.rule_type === 'ContainsAtLeastOneOf') {
+                      handledAnswersArray[choiceIndex] = true;
+                    } else if (ruleSpec.rule_type ===
+                      'DoesNotContainAtLeastOneOf') {
+                      handledAnswersArray = newHandledAnswers(choiceIndex);
```

Comment:
I'm not sure you need an entirely new function here. The following should suffice?

```
for (var i = 0; i < handledAnswersArray.length; i++) {
  if (i !== choiceIndex) {
    handledAnswersArray[i] = true;
  }
}
```